### ISSUE TITLE
[quoted-strings] false positive on flow-mapping values starting with ':'

### ISSUE THREAD

**Issue by jsnb-devoted** [adrienverge/yamllint#818]
## Summary

With `quoted-strings: {required: only-when-needed}`, yamllint flags a double-quoted string starting with `:` inside a **flow mapping** as redundantly quoted — but removing the quotes produces a real YAML parse error.

## Environment

- yamllint 1.38.0 (also reproduced on older 1.x)
- Python 3.12, PyYAML 6.x

## Reproduction

`.yamllint`:
```yaml
rules:
  quoted-strings:
    quote-type: double
    required: only-when-needed
```

`test.yaml`:
```yaml
foo:
  within: {dimension: email_created_at, last: {days: ":days"}}
```

```
$ yamllint test.yaml
test.yaml
  2:54  error  string value is redundantly quoted with double quotes  (quoted-strings)
```

## Expected behavior

The quotes are **required**. Removing them yields invalid YAML:

```
$ python3 -c 'import yaml; yaml.safe_load("foo: {days: :days}")'
yaml.parser.ParserError: while parsing a flow node
expected the node content, but found ':'
```

The leading `:` inside a flow mapping is interpreted as a flow-value indicator, so the scalar must be quoted. yamllint's own parser agrees — it reports `syntax error: expected the node content, but found ':'` on the unquoted form.

## Actual behavior

yamllint reports the quoted form as redundantly quoted, forcing users to choose between a lint error and a parse error.

## Related

#799 (self-closed as "already fixed") covered *trailing*-colon strings in **block** context (`- text: "foo:"`). This report is a distinct case: *leading* colon in **flow-mapping** context.

## Workaround

Rewrite the mapping in block style — a value of `:days` parses fine unquoted outside a flow mapping:

```yaml
foo:
  within:
    dimension: email_created_at
    last:
      days: :days
```
